Frontend/node/default.nix

```diff
@@ -0,0 +1,50 @@
+{ inputs, lib, flake-parts-lib, ... }:
+let
+  inherit (flake-parts-lib)
+    mkPerSystemOption;
+  inherit (lib)
+    types;
+in
+{
+  options = {
+    perSystem = mkPerSystemOption
+      ({ config, self', lib, pkgs, system, ... }:
+        let
+          nodePackages = inputs.dream2nix.lib.makeFlakeOutputs {
+            systems = [ system ];
+            source = ./.;
+            projects = {
+              "atlas-ui" = {
+                name = "atlas-ui";
+                subsystem = "nodejs";
+                translator = "package-lock";
+                builder = "strict-builder";
+                subsystemInfo = {
+                  nodejs = 14;
+                };
+              };
+            };
+          };
+
+          nodeDependencies = nodePackages.packages.${system}.atlas-ui.lib;
+        in
+        {
+          options.nammayatri = lib.mkOption {
+            type = types.submodule {
+              options = {
+                nodeDependencies = lib.mkOption {
+                  type = types.package;
+                };
+                nodejs = lib.mkOption {
+                  type = types.package;
+                };
+              };
+            };
+          };
+          config.nammayatri = {
+            inherit nodeDependencies;
+            nodejs = pkgs.nodejs-14_x;
+          };
```

Comment:
Just put this in `config.packages.nodeDependencies` (or named suitably) and let it build as part of CI. We don't need a submodule here.

Reply:
And instead of `nodejs`, expose a devShell containing the node binary and [compose](https://haskell.flake.page/devshell#composing-devshells) it at higher level.

Reply:
how does this proposed devShell solution work for the bundle derivation? I can probably change the watch script to rely on being run in the devShell and add the node_path that way but I am unclear how to utilize this in a non-shell derivation.

Reply:
The bundle drv wouldn't use the devShell. Instead of `config.nammayatri.nodejs` it would use `pkgs.nodejs-14_x` (or `self'.packages.nodejs`).